app/web/features/search/SearchPage.tsx

```diff
@@ -52,193 +55,195 @@ const useStyles = makeStyles((theme) => ({
+    name: locationName,
+    simplifiedName: locationName,
+  });
+  const [queryName, setQueryName] = useState<undefined | string>(undefined);
+  const [searchType, setSearchType] = useState("location");
+  const [lastActiveFilter, setLastActiveFilter] = useState(0);
+  const [hostingStatusFilter, setHostingStatusFilter] = useState(0);
+  const [numberOfGuestFilter, setNumberOfGuestFilter] = useState(undefined);
+  const [completeProfileFilter, setCompleteProfileFilter] = useState(true);
   const [selectedResult, setSelectedResult] = useState<
     Pick<User.AsObject, "username" | "userId" | "lng" | "lat"> | undefined
   >(undefined);
-
-  const previousResult = usePrevious(selectedResult);
-
-  const [areClustersLoaded, setAreClustersLoaded] = useState(false);
-
-  const showResults = useRef(false);
-
-  const searchFilters = useRouteWithSearchFilters(searchRoute);
-
-  const updateMapBoundingBox = (
-    newBindingBox: [number, number, number, number] | undefined
-  ) => {
-    if (newBindingBox && newBindingBox.join() !== "0,0,0,0") {
-      map.current?.fitBounds(newBindingBox);
-    }
-  };
-
-  useEffect(() => {
-    if (showResults.current !== searchFilters.any) {
-      showResults.current = searchFilters.any;
-      setTimeout(() => {
-        map.current?.resize();
-      }, theme.transitions.duration.standard);
-    }
-  }, [searchFilters.any, selectedResult, theme.transitions.duration.standard]);
-
-  const flyToUser = useCallback((user: Pick<User.AsObject, "lng" | "lat">) => {
-    map.current?.stop();
-    map.current?.easeTo({
-      center: [user.lng, user.lat],
-    });
-  }, []);
-
-  useEffect(() => {
-    if (previousResult) {
-      //unset the old feature selection on the map for styling
-      areClustersLoaded &&
-        map.current?.setFeatureState(
-          { source: "clustered-users", id: previousResult.userId },
-          { selected: false }
+  const [isFiltersOpen, setIsFiltersOpen] = useState(false);
+  const [mapInitiallyLocated, setMapInitiallyLocated] = useState(false);
+
+  // Loads the list of users
+  const { data, error, isLoading, fetchNextPage, isFetching, hasNextPage } =
+    useInfiniteQuery<UserSearchRes.AsObject, Error>(
+      [
+        "userSearch",
+        queryName,
+        locationResult?.name,
+        locationResult?.bbox,
+        lastActiveFilter,
+        hostingStatusFilter,
+        numberOfGuestFilter,
+        completeProfileFilter,
+      ],
+      ({ pageParam }) => {
+        const lastActiveComparation = parseInt(
+          lastActiveFilter as unknown as string
         );
-    }
-
-    if (selectedResult) {
-      //update the map
-      flyToUser(selectedResult);
-      areClustersLoaded &&
-        map.current?.setFeatureState(
-          { source: "clustered-users", id: selectedResult.userId },
-          { selected: true }
+        const hostingStatusFilterComparation = parseInt(
+          hostingStatusFilter as unknown as string
```

Comment:
Same here with unknown. I think if you set the type in the `useState` variable up top you won't need to do this here.

Reply:
yeah all this part it's a bit dirty code, I just did it to make it work and promise myself to improve it later, I might change this part in the future since I plan to implement pagination instead of the infinitive scroll, I was thinking about doing that refactor there (since this PR it's getting to big already, but I fully agree on improving this part of the code)

Reply:
I think if you just set the types of the useState variables you can remove all of these at once. TS is just being anxious it doesn't know what these are.